src/NetworkChest.lua

```diff
@@ -609,6 +609,62 @@ function M.onTick()
   M.update_queue()
 end
 
+function M.onTick_60()
+  M.updatePlayers()
+  M.check_alerts()
+end
+
+function M.handle_missing_material(entity, name)
+  -- did we already transfer something for this ghost/upgrade?
+  if GlobalState.alert_transfer_get(entity.unit_number) ~= true then
+    -- We can only do something about entities in a network
+    -- REVISIT: assuming "player" force only
+    local net = entity.surface.find_logistic_network_by_position(entity.position, "player")
+    if net ~= nil and net.available_construction_robots > 0 then
+      local network_count = GlobalState.get_item_count(name)
+      if network_count > 0 then
+        local n_inserted = net.insert({name=name, count=1})
```

Comment:
Why only insert one item?

Reply:
Each alert is for one missing item, which points to one entity.

I track "handled" alerts by recording the entity.unit_number. I don't want to send an item twice for the same ghost.

I could combine all outstanding alerts and request a bunch at the same time, but that gets complicated.
How would I track which alerts have been handled?

The code has to handle the case where I add an item to the network, but it gets used for a different purpose.
That's what the timeout is for.

Reply:
After a second read-through this makes sense. Those transfers should be cheap, but might need revisiting if performance somehow becomes an issue if players are constructing a ton of entities at once.